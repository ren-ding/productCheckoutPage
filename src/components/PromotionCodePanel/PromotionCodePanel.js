import React, {Component} from 'react';
import PropTypes from 'prop-types';

export default class PromotionCodePanel extends Component {
    render() {
        let input;
        return (
            <div className='ui action labeled input'>
                <div className="ui label">
                    Promotion Code:
                </div>
                <input type="text" ref = { node => {
                    input = node;
                }}
                />
                <button className='ui teal button code-submit' onClick={() => this.props.submitPromotionCode(input.value)}>Submit Code</button>
                {this.renderCodeMessage()}
            </div>
        );
    }

    renderCodeMessage = () => {
        if (this.props.codeMessage) {
            return (
                <div className='ui purple tag label'>
                    {this.props.codeMessage}
                </div>
            );
        }
    }
}

PromotionCodePanel.propTypes = {
    submitPromotionCode: PropTypes.func.isRequired,
    codeMessage: PropTypes.string.isRequired
};