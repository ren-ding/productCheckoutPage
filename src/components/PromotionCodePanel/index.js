import PromotionCodePanel from './PromotionCodePanel';
export default PromotionCodePanel;