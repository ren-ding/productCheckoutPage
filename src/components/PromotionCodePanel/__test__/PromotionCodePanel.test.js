import React from 'react';
import PromotionCodePanel from '../PromotionCodePanel';
import renderer from 'react-test-renderer';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });

describe('PromotionCodePanel', ()=> {
    let promotionCodePanel;
    let props;

    beforeEach(()=> {
        props = {
            submitPromotionCode: ()=>{},
            codeMessage:""
        }

        promotionCodePanel = () => mount(<PromotionCodePanel {...props} />);        
    });

    describe('render', () => {
       it('should render to match snapshot', () => {
           const tree = renderer.create(<PromotionCodePanel {...props}/>).toJSON();
           expect(tree).toMatchSnapshot();
       }); 
    });

    describe('click submit code button', ()=> {
        it('should call submitPromotionCode event handler', ()=> {
            const mockSubmitPromotionCode = jest.fn();
            props = {
                submitPromotionCode: mockSubmitPromotionCode,
                codeMessage:""
            }
            const panel = promotionCodePanel();
            //reason for this panel using enzyme mount rather than shallow is that
            //the component contains an input cached by ref and used for click event handler 
            //shallow has no facility to mock this up
            panel.find('.code-submit').simulate('click');
            expect(mockSubmitPromotionCode).toHaveBeenCalled();
        });
    });
});