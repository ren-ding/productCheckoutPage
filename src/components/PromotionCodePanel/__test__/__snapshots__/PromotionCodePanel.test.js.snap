// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromotionCodePanel render should render to match snapshot 1`] = `
<div
  className="ui action labeled input"
>
  <div
    className="ui label"
  >
    Promotion Code:
  </div>
  <input
    type="text"
  />
  <button
    className="ui teal button code-submit"
    onClick={[Function]}
  >
    Submit Code
  </button>
</div>
`;
